iga_1D.py

```diff
@@ -1,127 +1,245 @@
 import numpy as np
 from scipy import linalg as la
 import matplotlib.pyplot as plt
+
 import pca_01 as pca
 import curveFitting as cfit
+import plottingScripts as plts
 
-def scalingSegment(ua,ub):
-    localpts = np.zeros(len(gaussLegendrePoints))
+def scalingSegment(ua,ub,gausspt):
+    localpts = np.zeros(len(gausspt))
     for i in range(len(localpts)):
-        localpts[i] = 0.5*(ub - ua)*gaussLegendrePoints[i] + 0.5*(ub + ua)
+        localpts[i] = 0.5*(ub - ua)*gausspt[i] + 0.5*(ub + ua)
     return localpts
 
-def localStiffnessMatrix(intgpt):
+def jacobian(U,p,u,PGeom):
+    derNB = pca.derNFunction(U,p,u)
+    jcb = derNB@PGeom[0].T
+    return jcb
+
+################ WEAK FORM INTEGRALS ####################
+
+def localStiffnessMatrix(U,p,intgpt):
     derNB = pca.derNFunction(U,p,intgpt)
-    lke = derNB.transpose()*E@derNB
+    lke = derNB.T*E@derNB
     return lke
 
-def localForceVector(intgpt):
+def localForceVector(U,p,intgpt,xpt):
     nB = pca.nFunction(U,p,intgpt)
-    if intgpt < 5.0:#Be careful with this condition
+    if xpt < 0.5*L:#Be careful with this condition
```

Comment:
why?

Reply:
When I was developing the IGA code for 1D axial stress, I had some issues regarding the value of some integration points. The problem stated a value of the density of the beam at x >= 5.0. Initially I considered that the value of the integration points would be the one needed for enforcing different values of density, however, I was wrong. That value of x >= 5.0 was a geometric condition, so I had to calculate the true point and if that coordinate was higher than 5.0, the change of density shall be applied.

If you have more doubts, I encourage you to write down here so that I can review the code as well and give yo a more accurate answer